treewide: replace defunct git://github.com urls with https://

lucc opened the pull request:
###### Description of changes

Fix #165246

Github has removed support for git://github.com and recommends https://github.com instead.



###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

lucc committed:
treewide: replace defunct git://github.com urls with https://

Github has removed support for git://github.com and recommends
https://github.com instead.

See https://github.blog/2021-09-01-improving-git-protocol-security-github/

lucc committed:
docs: replace defunct git://github.com urls with https://

Artturin commented:
Meta.repositories changed don't need to be in this https://github.com/NixOS/nixpkgs/pull/165512

lucc commented:
I rebased and got rid of the meta.repository changes.

I also tried to run some of the generate.sh scripts but they also updated some (node) dependencies apart from the url change. @matejc you are maintaining nixui, do you think that is ok?

I could not find the maintainer of onlykey (`git grep maintainer pkgs/tools/security/onlykey` returns nothing). But the same problem exists there.

lucc committed:
luarocks-packages: replace git://github.com with https://

lucc commented:
For matrix-appservice-discord the ./generate.sh script failes because upstreams specifies some dependencies with git://giithub.com links.
@pacien are you involved in the upstream and can easily fix that?

lucc commented:
frab also references git://github.com but it seems that frab was removed in 6fa3728805f04ec367b8f42ba3d3fc77e3d96513 a long time ago. @Mic92 can you shed some light on why the folder `pkgs/servers/web-apps/frab/` and its contents exist even though you deleted the default.nix in 6fa3728805f04ec367b8f42ba3d3fc77e3d96513?

lucc commented:
I did update the url in maintainers/scripts/luarocks-packages.csv but did not run maintainers/scripts/update-luarocks-packages because that updates the version of many lua packages and still leaves the homepage for luafilesystem in pkgs/development/lua-modules/generated-packages.nix  with a git:// url. (see 51b5c740a7d31d95924b82c20959e2210231fbe2 for example).

pacien commented:
@lucc wrote:
> For matrix-appservice-discord the ./generate.sh script failes because
> upstreams specifies some dependencies with git://giithub.com links. @pacien
> are you involved in the upstream and can easily fix that?

Upstream PR: https://github.com/Half-Shot/matrix-appservice-discord/pull/777

This however is very unlikely to get merged soon, as the project has been
inactive for quite a while.

kalbasit commented:
Assigning myself the pr, I can update onlykey and test it.

SuperSandro2000 commented:
@lucc you can easily revert the onlykey changes and just manually replace line 6351.

Artturin commented:
@Stunkymonkey would you be interested in switching some of these over to `fetchFromGithub`? :slightly_smiling_face:

jtojnar commented on pkgs/applications/video/byzanz/default.nix:
```diff
@@ -5,7 +5,7 @@ stdenv.mkDerivation {
 
   src = fetchgit {
-    url = "git://github.com/GNOME/byzanz";
+    url = "https://github.com/GNOME/byzanz";
```
Does not exist, should go to `https://gitlab.gnome.org/Archive/byzanz`

SuperSandro2000 replied:
```suggestion
    url = "https://gitlab.gnome.org/Archive/byzanz";
```

SuperSandro2000 committed:
byzan: switch to working source

Co-authored-by: Jan Tojnar <jtojnar@gmail.com>

SuperSandro2000 committed:
nixui: replace git://github.com with https://

SuperSandro2000 committed:
onlykey: replace git://github.com with https://

SuperSandro2000 reviewed (approved):
I dropped all changes to nodePackages which means this should be no-opt now.

LGTM, going to merge when ofborg is done since it is only a protocol change and nothing about the hashes should change.

lucc commented:
I am sorry, I was AFK over the weekend. Thanks @SuperSandro2000 for picking this up.

SuperSandro2000 commented:
No problem. Everyone has a life and other important things to do.
